### PR TITLE
[wip] Add Twitter Model Architecture Summary

### PR DESCRIPTION
**Pull request by sidneyfletcher**
[Rendered](https://github.com/sidneyfletcher/awesome-twitter-algo/blob/main/masknet.md)

Hey this is kinda a first draft overview of what MaskNet is (I didn't know so I read the paper). Lemme know what you think!

### CODE REVIEW

**Review by veekaybee — COMMENTED** [review]
😮 This is really cool! I left a few small notes which might make the text clearer. Could you also add a a link in the main README to where you'd like this to go? Something else which might be cool if you have time (only if you have time) is links to any [relevant code in the repo ](https://github.com/twitter/the-algorithm-ml/tree/main/projects/home/recap) that ties back to this, for example it would be cool to know how [these features](https://github.com/twitter/the-algorithm-ml/blob/main/projects/home/recap/FEATURES.md) get crossed in the code

File: masknet.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+In particular, $\mathbf{W}$ is a symmetric matrix with $w_{ij} = w_{ji}$ where each element is the weight for the interaction between feature $i$ and feature $j$. There are $n(n-1)/2$ unique elements in this matrix.
+
+## Factorization Machines
+
+In a Factorization Machine (FM), we factorize the weights matrix $\mathbf{W}$ into a matrix decomposition $\mathbf{V} \mathbf{V}^T$.
+
+$$\hat{y} = \beta_0 + \mathbf{x} \mathbf{\beta} + \mathbf{x} \mathbf{V} \mathbf{V}^T \mathbf{x}^T$$
+
+where $\mathbf{V}$ is a matrix of latent factors, or to use the terminology from NLP, embeddings. (One of the key insights of NLP is that training Word2Vec embeddings is equivalent to matrix factorization).
```
[the remark this answers is not in the document]
Comment (sidneyfletcher) [reply masknet.md]: Yup! this paper kinda blew my mind when I read it tbh

File: masknet.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
+# Twitter Model Architecture Summary
+
+## Summary
+
+Because of the high number of categorical features and important interaction effects, Ads/RecSys models often use models which specifically incorporate cross features. We review a couple different approaches to this problem and explain how MaskNet, the model used by the Twitter heavy ranker, works.
```
[the remark this answers is not in the document]
Comment (sidneyfletcher) [reply masknet.md]: I sorta addressed this and fleshed it out a bit in the background

**Review by veekaybee — APPROVED** [review]

**Comment by veekaybee** [comment]
Thanks!!

**Pull request merged**